pilot/pkg/networking/core/networkfilter.go

```diff
@@ -74,6 +77,23 @@ func setAccessLogAndBuildTCPFilter(push *model.PushContext, node *model.Proxy,
 	return tcpFilter
 }
 
+func buildSNIDFPFilter(port int, svc *model.Service) *listener.Filter {
+	sniDfp := &snidfp.FilterConfig{
+		DnsCacheConfig: &dfp.DnsCacheConfig{
+			Name:            model.BuildDNSCacheName(svc.Hostname),
```

Comment:
Will there be errors if this cache conflicts with HTTP caches?

Reply:
Based on e[nvoy's DNS cache documentation](https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/common/dynamic_forward_proxy/v3/dns_cache.proto#envoy-v3-api-msg-extensions-common-dynamic-forward-proxy-v3-dnscacheconfig), a cache conflict will only result in an error if the if the caches will the same name have different configs. In our case, all dns cache configs are the same. It would be safer to add port to the dns cache name to prevent conflict/errors.